src/main/java/com/lacunasoftware/pkiexpress/Authentication.java

```diff
@@ -77,6 +78,10 @@ public void setSignature(String signatureBase64) {
 		this.signature = signatureBase64;
 	}
 
+	public void setSignature(byte[] signature) {
+		this.signature = new String(signature);
+	}
+
```

Comment:
Here, you are converting a byte[] into a hexadecimal string, it's not what PKI Express supports. You should convert this byte[] into a base64 string. Something like the code below:

```java
this.signature = Base64.getEncoder().encodeToString(signature);
```